tt_eager/tt_dnn/op_library/unpad/unpad_op.cpp

```diff
@@ -19,17 +19,80 @@ namespace tt {
 namespace tt_metal {
 
 
+uint32_t get_upper_dims_compressed(const Tensor & tensor) {
+
+    uint32_t upper_dims_compressed = 1;
+    for(int dim=0; dim<tensor.shape().rank() - 2; dim++) {
+        upper_dims_compressed *= tensor.shape()[dim];
+    }
+    return upper_dims_compressed;
+}
+
+uint32_t get_upper_start_offset(const Tensor & tensor, const Shape & output_tensor_start) {
+
+    uint32_t num_pages = tensor.volume();
+    if(tensor.layout() == Layout::TILE) {
+        num_pages /= (TILE_HW);
+    }
+    else {
+        uint32_t page_width = tensor.shape()[-1];
+        num_pages /= page_width;
+    }
+    // offset for every dim except last 2
+    uint32_t start_offset = 0;
+    for(int dim_outer=0; dim_outer<tensor.shape().rank() - 2; dim_outer++) {
+        uint32_t compressed_dims = 1;
+        for(int dim_inner = dim_outer; dim_inner >= 0; dim_inner--) {
+            compressed_dims*=tensor.shape()[dim_inner];
+        }
+        start_offset += (num_pages/compressed_dims) * output_tensor_start[dim_outer];
+    }
+    return start_offset;
+
+}
+
+
+
+uint32_t get_tiled_start_offset(const Tensor &input_tensor,
+                        const Shape &output_tensor_start
+ ) {
+    uint32_t num_input_pages = input_tensor.volume() / (TILE_HW);
+
+    uint32_t upper_dims_compressed = get_upper_dims_compressed(input_tensor);
+    uint32_t num_pages_width = num_input_pages / (upper_dims_compressed * (input_tensor.shape()[-2]/TILE_HEIGHT));
+
+
+    // offset for every dim except last 2
+    uint32_t start_offset = get_upper_start_offset(input_tensor, output_tensor_start);
+
+    start_offset += output_tensor_start[-2]/TILE_HEIGHT*num_pages_width
+                                + output_tensor_start[-1]/TILE_WIDTH;
+    return start_offset;
+
+}
+
+uint32_t get_rm_start_offset(
+                        const Tensor &tensor,
+                        const Shape &output_tensor_start
+ ) {
+
+    uint32_t num_pages = tensor.volume() / tensor.shape()[-1];
+    uint32_t upper_dims_compressed = get_upper_dims_compressed(tensor);
+    uint32_t num_pages_width = num_pages / (upper_dims_compressed * (tensor.shape()[-2]));
```

Comment:
Isnt this value always just 1?

Reply:
yea it is , haha its copy paste from the code in tiled where it can be multiple page widths. I'll change that.